core/src/listener/listener.ts

```diff
@@ -204,7 +209,7 @@ function latestJob({
     try {
       // We assume that redis cache has been initialized within
       // `State.add` method call.
-      observedBlock = Number(await redisClient.get(observedBlockRedisKey))
+      observedBlock = (await getObservedBlock({ service })).blockNumber
```

Comment:
aren't redis utilized anymore? it'd be nice if it first tries redis and fallback to db

Reply:
I was discussing about this with Intizar, but we decided to go first with DB approach and then after benchmarks to think about Redis. The issue is that previously Redis was not used as cache but as database which makes it confusing to move it to cache directly, therefore focusing on DB first and then adding Redis as cache later.

Reply:
good point Nick. I've decided to remove redis in the PR and try to include it in the future for optimizing performance. The logic was getting more complex for a single PR if I wanted to optimize with redis